### PR TITLE
Character lab collection fields

### PR DESCRIPTION
**Pull request by jwalz**
## Purpose
Add two new fields for Collections and CollectionSubmissions (`school_type` and `study_design`) to support Character Lab workflows

## Changes
* Adds the two new fields to the Collection and CollectionSubmissions (the Collection fields allow admins to define the choices available for the submissions)
* Admin App functionality for setting up the new fields
* Stop setting up a `primary_collection` upon creation of `RegistrationProvider`s. This feature was unused and its implementation meant that no new fields could be added to Collections without breaking migrations
* Fix an assumption in the Admin App Delete API that RegistrationProviders have a `primary_collection`
* Remove API tests for listing CollectionSubmissions on a RegistrationProvider
*  Fix another migration that directly imported the Collection model directly instead of working on the historical model
* Fix an unrelated, broken Celery import

## QA Notes
Verify that Collections can be created to support the new `school_type` and `study_design` fields

### CODE REVIEW

**Comment by brianjgeiger** [comment]
Is there anything that needs to be done for OSF Search to get these new fields, or does it just put everything on the model into search?

**Comment by jwalz** [comment]
> Is there anything that needs to be done for OSF Search to get these new fields, or does it just put everything on the model into the index?

Good catch!

**Review by brianjgeiger — APPROVED** [review]
Good catch on the API serialization. That would have been more difficult to use without it.

File: osf/utils/migrations.py [hunk 1]
```diff
@@ -374,17 +374,18 @@ def create_schema_blocks_for_question(state, rs, question, sub=False):
...
-        create_schema_block(
-            state,
-            rs.id,
-            block_type,
-            required=question.get('required', False),
-            schema_block_group_key=schema_block_group_key,
-            registration_response_key=get_registration_response_key(question)
-        )
+        if question.get('format') or question.get('type'):
+            # Creates question input block - this block will correspond to an answer
```
Comment (brianjgeiger) [inline osf/utils/migrations.py:378]: Is this change meant to be here?

Comment (jwalz) [reply osf/utils/migrations.py]: nope, thought I stashed this correctly, thanks for the catch

**Pull request merged**